redisvl/index/index.py

```diff
@@ -814,8 +837,21 @@ def batch_query(
             all_parsed.append(parsed)
         return all_parsed
 
+    def _validate_query(self, query: BaseQuery) -> None:
+        """Validate a query."""
+        if isinstance(query, VectorQuery):
+            field = self.schema.fields[query._vector_field_name]
+            if query.ef_runtime and field.attrs.algorithm != "hnsw":  # type: ignore
```

Comment:
I think there's an enum for this

Reply:
I'll find that. Also, this error message must have been an autocomplete.